metadata-ingestion/src/datahub/ingestion/source/ldap.py

```diff
@@ -135,13 +137,41 @@ class LDAPSourceConfig(StatefulIngestionConfigBase, DatasetSourceConfigMixin):
 
     manager_pagination_enabled: bool = Field(
         default=True,
-        description="Use pagination while search for managers (enabled by default).",
+        description="[deprecated] Use pagination_enabled ",
+    )
+    _deprecate_manager_pagination_enabled = pydantic_field_deprecated(
```

Comment:
Why is this required ? There is also manager_filter_enabled. Does it also need renaming ?

Suggestion : Use pydantic_renamed_field instead. No need to write your own validator in that case.

Reply:
> Why is this required ?

LDAP service like database, we can get error "SIZELIMIT_EXCEEDED: {'desc': 'Size limit exceeded'}"
This error tells us that we reached the limit of search result entries (1000 by default). There are 2 ways of resolving this error: 
- Go to your LDAP DATABASE and increase the maximum page size (MaxPageSize) option. 
- Use result paginating in your Python code.

Also, LDAP DATABASE sometimes doesn't support pagination at all. For example, google workspace LDAP doesn't support it -> so we need the option to turn off pagination completely, or we get an error.

Sometimes finding a person manager provides unexpected data and can be reduced with pagination.

> There is also manager_filter_enabled. Does it also need renaming?

No, this option is responsible for additional manager discovery. We want that person's head will be founded or not.

> Suggestion : Use pydantic_renamed_field instead. No need to write your own validator in that case.

Thx, I change it.

Reply:
Right. My question was around - why this rename is required - manager_pagination_enabled->pagination_enabled. Does this rename also warrant a rename in manager_filter_enabled->filter_enabled ?

Reply:
In the old code/way, we always setup page control with line `self.lc = create_controls(self.config.page_size)`.
Sometimes LDAP provider doesn't support pagination. That's why i need an option – enable pagination or not in new code.

Now I have a decision:
1) create a new variable, for example `pagination_enabled` and `save manager_pagination_enabled`
2) rename `manager_pagination_enabled` to `pagination_enabled`

I choose option 2) because I don't see a situation where we need pagination for manager discovery, and at the same time, we need a global user/group pagination. If we have a problem and pagination is needed for the user/group or head, we can enable or disable it globally. The user will need clarification, as I am, with the name manager_pagination_enabled and global enable/disable pagination.